pkg/auth/scope/scope.go

```diff
@@ -56,5 +56,5 @@ func VerifyScope(ctx context.Context, scopeMap map[string]*authpb.Scope, resourc
 }
 
 func hasRoleEditor(scope authpb.Scope) bool {
-	return scope.Role == authpb.Role_ROLE_EDITOR
+	return scope.Role == authpb.Role_ROLE_OWNER || scope.Role == authpb.Role_ROLE_EDITOR || scope.Role == authpb.Role_ROLE_UPLOADER
```

Comment:
I think it's better create two functions similar to this, one for checking the editor role and the other for uploader, and than while checking the publicshareScope for the `InitiateFileUploadRequest` check the hasRoleUploader also.

This also require that this function https://github.com/cs3org/reva/blob/master/internal/grpc/interceptors/auth/scope.go#L218 is aware of the uploader role (as is aware of the editor role)